nautobot/ipam/factory.py

```diff
@@ -111,3 +99,85 @@ def prefix(self, n):
             return f"2003:0{n:X}::/32"
 
         raise RuntimeError(f"Don't know how to pick an address for RIR {self.rir.name}")
+
+
+def random_route_distinguisher():
+    """
+    The RD is an arbitrary unique string up to VRF_RD_MAX_LENGTH (21) characters with no explicit validation.
+
+    However by convention it will be one of:
+    - "<2-byte ASN>:<4-byte integer>"
+    - "<IPv4 address>:<2-byte integer>"
+    - "<4-byte ASN>:<2-byte integer>"
+    """
+    branch = randgen.randint(0, 2)
+    if branch == 0:
+        # 16-bit ASNs 64496–64511 are reserved for documentation and sample code
+        return f"{randgen.randint(64496, 64511)}:{randgen.randint(0, 2**32 - 1)}"
+    if branch == 1:
+        # The official stance of factory-boy maintainers is that evaluate() is a private method, but there doesn't
+        # seem to be any convenient alternative way to do this at present.
+        return f"{factory.Faker('ipv4_private').evaluate(None, None, {'locale': None})}:{randgen.randint(0, 2**16 - 1)}"
+    # 32-bit ASNs 4200000000-4294967294 are reserved for private use
+    return f"{randgen.randint(4200000000, 4294967294)}:{randgen.randint(0, 2**16 - 1)}"
+
+
+class RouteTargetFactory(PrimaryModelFactory):
+    class Meta:
+        model = RouteTarget
+        exclude = (
+            "has_description",
+            "has_tenant",
+        )
+
+    # Name needs to be globally unique, but the random route-distinguisher generation space is large enough that
+    # we'll deal with collisions as and when they occur.
+    name = factory.LazyFunction(random_route_distinguisher)
+
+    has_description = factory.Faker("pybool")
+    description = factory.Maybe("has_description", factory.Faker("sentence"), "")
+
+    has_tenant = factory.Faker("pybool")
+    tenant = factory.Maybe("has_tenant", random_instance(Tenant), None)
+
+
+class VRFFactory(PrimaryModelFactory):
+    class Meta:
+        model = VRF
+        exclude = (
+            "has_description",
+            "has_rd",
+            "has_tenant",
+        )
+
+    # note that name is *not* globally unique
+    name = factory.Faker("color_name")
+
+    # RD needs to be globally unique, but the random route-distinguisher generation space is large enough that
```

Comment:
Are these are deterministically randomized so they're the same every time tests are run? So we shouldn't need to be concerned about this unless we change the seed or number of objects generated?

Reply:
Or we make a change to the factories that results in a different path through the RNG. But yes, if no changes in any of thosse areas, we're supposed to get the same instances every time.